File: src/models/task.ts
/** Types cho tính năng AI Tasks / AI Cowork — khớp với DTO ở `src-tauri/src/models/task.rs`. */

export type TaskCategory = "screen" | "batch" | "part" | "other";

export const TASK_CATEGORY_OPTIONS: { label: string; value: TaskCategory }[] = [
  { label: "Screen", value: "screen" },
  { label: "Batch", value: "batch" },
  { label: "Part", value: "part" },
  { label: "Other", value: "other" },
];

export const TASK_CATEGORY_META: Record<TaskCategory, { badgeClass: string }> = {
  screen: { badgeClass: "bg-sky-100 text-sky-700" },
  batch: { badgeClass: "bg-violet-100 text-violet-700" },
  part: { badgeClass: "bg-amber-100 text-amber-700" },
  other: { badgeClass: "bg-canvas text-muted" },
};

export type WfProcStepStatus = "pending" | "in_progress" | "completed" | "skipped";

export const STEP_STATUS_META: Record<WfProcStepStatus, { badgeClass: string }> = {
  pending: { badgeClass: "bg-canvas text-muted" },
  in_progress: { badgeClass: "bg-sky-100 text-sky-700" },
  completed: { badgeClass: "bg-emerald-100 text-emerald-700" },
  skipped: { badgeClass: "bg-canvas text-muted" },
};

export interface Task {
  id: number;
  task_cd: string;
  task_name: string;
  category_id: string;
  is_complete: boolean;
  completed_at: string;
  created_at: string;
  created_by: string;
  updated_at: string;
  updated_by: string;
  /** Tên workflow đang áp dụng cho step in_progress gần nhất — chỉ có ở kết quả `taskList`. */
  current_wf_name: string;
  /** Tên step in_progress gần nhất — chỉ có ở kết quả `taskList`. */
  current_step_name: string;
  /** Trạng thái step in_progress gần nhất — chỉ có ở kết quả `taskList`. */
  current_step_status: string;
}

export interface CreateTaskRequest {
  task_cd: string;
  task_name: string;
  category_id: string;
}

export interface UpdateTaskRequest {
  task_cd: string;
  task_name: string;
  category_id: string;
  is_complete: boolean;
}

export interface TaskWfProc {
  id: number;
  task_id: number;
  wf_id: number;
  latest_step_id: number | null;
  created_at: string;
  created_by: string;
  updated_at: string;
  updated_by: string;
}

export interface CreateWfProcRequest {
  task_id: number;
  wf_id: number;
}

export interface TaskWfProcStep {
  id: number;
  wf_proc_id: number;
  wf_step_id: number;
  status: string;
  created_at: string;
  created_by: string;
  updated_at: string;
  updated_by: string;
}

export interface CreateWfProcStepRequest {
  wf_proc_id: number;
  wf_step_id: number;
  status: string;
}

File: src-tauri/src/models/task.rs
//! Model cho tính năng AI Tasks / AI Cowork — theo dõi một task đi qua từng
//! step của 1 workflow (bảng `tasks` + `task_wf_proc` + `task_wf_proc_step`).

use serde::{Deserialize, Serialize};

/// Một hạng mục công việc (bảng `tasks`).
#[derive(Clone, Debug, Serialize)]
pub struct Task {
    pub id: i32,
    pub task_cd: String,
    pub task_name: String,
    pub category_id: String,
    pub is_complete: bool,
    pub completed_at: String,
    pub created_at: String,
    pub created_by: String,
    pub updated_at: String,
    pub updated_by: String,
    /// Tên workflow đang áp dụng cho step in_progress gần nhất — chỉ có ở kết quả `task_list`.
    #[serde(default)]
    pub current_wf_name: String,
    /// Tên step in_progress gần nhất — chỉ có ở kết quả `task_list`.
    #[serde(default)]
    pub current_step_name: String,
    /// Trạng thái step in_progress gần nhất — chỉ có ở kết quả `task_list`.
    #[serde(default)]
    pub current_step_status: String,
}

#[derive(Debug, Deserialize)]
pub struct CreateTaskRequest {
    pub task_cd: String,
    #[serde(default)]
    pub task_name: String,
    #[serde(default)]
    pub category_id: String,
}

#[derive(Debug, Deserialize)]
pub struct UpdateTaskRequest {
    pub task_cd: String,
    #[serde(default)]
    pub task_name: String,
    #[serde(default)]
    pub category_id: String,
    #[serde(default)]
    pub is_complete: bool,
}

/// Một lượt "chạy" workflow cho 1 task (bảng `task_wf_proc`).
#[derive(Clone, Debug, Serialize)]
pub struct TaskWfProc {
    pub id: i32,
    pub task_id: i32,
    pub wf_id: i32,
    pub latest_step_id: Option<i32>,
    pub created_at: String,
    pub created_by: String,
    pub updated_at: String,
    pub updated_by: String,
}

#[derive(Debug, Deserialize)]
pub struct CreateWfProcRequest {
    pub task_id: i32,
    pub wf_id: i32,
}

/// Trạng thái của task tại 1 step cụ thể (bảng `task_wf_proc_step`).
#[derive(Clone, Debug, Serialize)]
pub struct TaskWfProcStep {
    pub id: i32,
    pub wf_proc_id: i32,
    pub wf_step_id: i32,
    pub status: String,
    pub created_at: String,
    pub created_by: String,
    pub updated_at: String,
    pub updated_by: String,
}

#[derive(Debug, Deserialize)]
pub struct CreateWfProcStepRequest {
    pub wf_proc_id: i32,
    pub wf_step_id: i32,
    #[serde(default)]
    pub status: String,
}
